python/python/glide/async_commands/core.py

```diff
@@ -1324,6 +1428,124 @@ async def zcount(
             ),
         )
 
+    async def zrange(
+        self,
+        key: str,
+        range_query: Union[RangeByIndex, RangeByLex, RangeByScore],
+        reverse: bool = False,
+    ) -> List[str]:
+        """
+        Returns the specified range of elements in the sorted set stored at `key`.
+
+        ZRANGE can perform different types of range queries: by index (rank), by the score, or by lexicographical order.
+
+        See https://redis.io/commands/zrange/ for more details.
+
+        Args:
+            key (str): The key of the sorted set.
+            range_query (Union[RangeByIndex, RangeByLex, RangeByScore]): The range query object representing the type of range query to perform.
+                - For range queries by index (rank), use RangeByIndex.
+                - For range queries by lexicographical order, use RangeByLex.
+                - For range queries by score, use RangeByScore.
+            reverse (bool): If True, reverses the sorted set, with index 0 as the element with the highest score
+                By default, `start` must be less than or equal to `stop` to return anything.
+                However, if range_query is not an index query, the `start` is the highest score to consider, and `stop` is the lowest score to consider,
+                therefore `start` must be greater than or equal to `stop` in order to return anything.
+
+        Returns:
+            List[str]: A list of elements within the specified range.
+            For idex queries, If `start` is greater than either the end index of the sorted set or `stop`, an empty list is returned.
+            For lex/score queries:
+                - If `start` is greater than either the end index of the sorted set or `stop` and `reverse` is set to False,
+                 an empty list is returned.
+                - If `stop` is greater than either the end index of the sorted set or `start` and `reverse` is set to True,
+                 an empty list is returned.
+            If `key` does not exist, it is treated as an empty sorted set, and the command returns an empty array.
+            If `key` holds a value that is not a sorted set, an error is returned.
+
+        Examples:
+            >>> await zrange("my_sorted_set", RangeByIndex(0, -1))
+                ['member1', 'member2', 'member3']  # Returns all members in ascending order.
+            >>> await zrange("non_existing_key", RangeByScore(ScoreBoundary(0), ScoreBoundary(-1)))
```

Comment:
lets skip the non existing key example and instead have a more complex example that gives more information, for example RangeByScore with InfBound.Negative, ScoreBoundary(3), limit=2